### ISSUE TITLE
Hang in Falcon40B when running with more layers

### ISSUE THREAD

**Issue by tt-aho** [tenstorrent-metal/tt-metal#5065]
Running for example the full falcon40b model on its own seems to hang in all_gather op. There seems to be some sort of L1 corruption on eth cores as this is only recoverable using reset, and not just reboot.

Some additional testing seems like after reset and reboot, the full 60 layer test can pass on the first run, but hangs when trying to run the test again.

If we run something like 1 layer, 2 layer, then 60 layer in the same test/process, it doesn't seem to hang after rerunning the test.

**Comment by aliuTT** [comment]
Repro on t3004 because this machine has the model weights cached:

- `git checkout aliu/falcon40b-debug`
- `export TT_METAL_ENV=dev && export TT_METAL_HOME=$PWD && export ARCH_NAME=wormhole_b0 && export PYTHONPATH=$TT_METAL_HOME && make build && source build/python_env/bin/activate`
- `make tests`
- `pytest -svv models/demos/falcon40b/tests/test_perf_falcon_v2.py::test_perf_bare_metal[BFLOAT8_B-SHARDED-falcon_40b-layers_60-decode_batch32]`

Model possibly leaves machine in weird state, following test is launching a kernel on the erisc to just read from dram. No ethernet usage. It looks like the erisc app is not getting launched, even though binaries are sent to `0x9040` and signal of ` is sent to `0x9004`. 

- `TT_METAL_SLOW_DISPATCH_MODE=1 ./build/test/tt_metal/unit_tests --gtest_filter="DeviceFixture.EthKernelsNocReadNoSend`

**Comment by davorchap** [comment]
The Falcon40 model runs in fast dispatch mode, and then the subsequent small test hangs in slow dispatch mode.

Does it also hang if you run it in fast dispatch mode?

**Comment by davorchap** [comment]
fyi @umadevimcw this is the eth hang issue.

**Comment by ubcheema** [comment]
This is what I understand of the status:
Running 1 or 2 layer multiple times back to back is fine.
Running 60 layer once is fine. 
Second run of 60 layer hangs.

**Comment by aliuTT** [comment]
@davorchap After running 16 layers ( I think any high layer number will repro), I cannot run any kernels on ethernet cores. Both fast dispatch unit tests and slow dispatch unit tests are failing.

**Comment by ubcheema** [comment]
I have diagnosed the issue to be due to a lingering BBE artifact in ethernet rom firmware.
BBE has the concept of epoch queues which are command queues sitting in device dram. In erisc fw <= 6.6, firmware reads in the write pointer from this BBE epoch q in dram to determine if erisc app firmware should be executed.
When I asked Allan to write 1 @0x9004 to trigger erisc app fw execution, I overlooked the fact that erisc will also look at the epoch q pointer in device dram.
The reason it worked so far is that the respective dram locations in TT-Metal would remain uninitialized random values. So, when firmware goes and reads in that dram value, its still not equal to 0, so base fw would execute app fw.
This remains true for all the (small/er) tests so far. So no issues in running tests back to back.
With 16 layer model, what happens is that the epoch q address that base fw is reading gets cleared to 0. I guess data or weights or something else populates those addresses. The first run is ok since its uninitialized. For second run the values are cleared to 0 (as part of the first run). So for the second run when base fw reads from epoch q in dram, it reads a write pointer of 0 and thinks there is nothing to do and does not execute app firmware.
The gist of it all is that base firmware does not need to look at epoch q in dram. Its the job of BBE erisc app/runtime firmware to interface with epoch queues. And Metal does not even have that concept of epoch queues, so base firmware should not read anything from dram on its own.
I have already removed dram/epoch q related operations from erisc base firmware from 6.8 onwards.
I flashed the latest fw (6.9) on t3004 and I am able to run ethernet unit test after 16 layer model. I can also run 16 layer model back to back. It does not hang anymore.
So the solution is to upgrade the boards to latest tt-flash which carries erisc 6.9.
